src/jwt/sign.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  DEFAULT_HASH_METHODS,
+  DEFAULT_SIGNATURE_METHOD,
+  createSignature,
+  encodeToBase64Url,
+  importKey,
+} from "./utils";
+import type { JWTRegisteredClaims } from "./types";
+
+const headerAlgMap: Record<string, string> = {
+  HMAC: "HS",
+  RSA: "RS",
+  ECDSA: "ES",
+};
+
+export async function signJWT<
+  T extends Record<string, any> = Record<string, any>
+>({
+  payload = {} as T,
+  secret,
+  issuer,
+  audience,
+  expires = 30,
+  signatureMethod = DEFAULT_SIGNATURE_METHOD,
+  hashMethod = DEFAULT_HASH_METHODS,
```

Comment:
- Could be nice if we export import object as a separate type/inteface export
- For defaults, we shall assign to a new object and not restructure the object